internal/controller/prefix_controller.go

```diff
@@ -81,6 +82,14 @@ func (r *PrefixReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		if controllerutil.ContainsFinalizer(prefix, PrefixFinalizerName) {
 			if !prefix.Spec.PreserveInNetbox {
 				if err := r.NetboxClient.DeletePrefix(prefix.Status.PrefixId); err != nil {
+					if errors.Is(err, utils.ErrNotFound) {
```

Comment:
If the Prefix is not found in NetBox, isn't this the expected target state so we should assume everything is fine and remove the finalizer? I propose to inverse the line 85 condition and remove line 93. wdyt @bruelea @MaIT-HgA ?

Reply:
I agree, it was a mistake from my end